docker-compose.yml

```diff
@@ -1,9 +1,61 @@
 services:
-  live-codes:
+  app:
     build:
       context: .
       dockerfile: Dockerfile
-    image: result/latest
+      args:
+        - SELF_HOSTED=true
+        - SELF_HOSTED_SHARE=${SELF_HOSTED_SHARE:-true}
+        - SELF_HOSTED_BROADCAST=${SELF_HOSTED_BROADCAST:-true}
+        - BROADCAST_PORT=${BROADCAST_PORT:-3030}
+        - SANDBOX_HOST_NAME=${SANDBOX_HOST_NAME:-${HOST_NAME:-livecodes.localhost}}
+        - SANDBOX_PORT=${SANDBOX_PORT:-8090}
+        - FIREBASE_CONFIG=${FIREBASE_CONFIG:-}
+        - DOCS_BASE_URL=${DOCS_BASE_URL:-null}
+    restart: unless-stopped
+    environment:
+      - SELF_HOSTED=true
+      - HOST_NAME=${HOST_NAME:-livecodes.localhost}
+      - PORT=${PORT:-443}
+      - SELF_HOSTED_BROADCAST=${SELF_HOSTED_BROADCAST:-true}
+      - BROADCAST_PORT=${BROADCAST_PORT:-3030}
+      - BROADCAST_TOKENS=${BROADCAST_TOKENS:-}
+      - SANDBOX_HOST_NAME=${SANDBOX_HOST_NAME:-${HOST_NAME:-livecodes.localhost}}
+      - SANDBOX_PORT=${SANDBOX_PORT:-8090}
+      - LOG_URL=${LOG_URL:-null}
+      - VALKEY_HOST=valkey
+      - VALKEY_PORT=6379
+    volumes:
+      - ./assets:/srv/build/assets
+
+  valkey:
```

Comment:
Is `valkey` a hard requirement for `app` to run? if so maybe add a `depends_on`

Reply:
Please check this (updated)

https://github.com/live-codes/livecodes/blob/9d7d7160537d95cea4a8c5308feb2904cd36cbfc/docker-compose.yml#L32-L42

also updated `restart: on-failure` to allow exiting if share is not enabled.

Reply:
You are right.
Now `app` `depends_on` `valkey`
This still works when `valkey` exits (when `SELF_HOSTED_SHARE` is set to `false`)